app/src/main/java/org/simple/clinic/sync/indicator/SyncIndicatorViewController.kt

```diff
@@ -33,43 +27,12 @@ class SyncIndicatorViewController @Inject constructor(
     private val dataSync: DataSync,
     @Named("frequently_syncing_repositories") private val frequentlySyncingRepositories: ArrayList<SynceableRepository<*, *>>
 ) : ObservableTransformer<UiEvent, UiChange> {
-
-  private val errorTypesToShowErrorFor = setOf(
-      NetworkRelated::class,
-      Unexpected::class,
-      ServerError::class
-  )
-
+  
   override fun apply(events: Observable<UiEvent>): ObservableSource<UiChange> {
     val replayedEvents = ReplayUntilScreenIsDestroyed(events)
         .replay()
 
-    return Observable.merge(
-        startSync(replayedEvents),
-        showPendingSyncStatus(replayedEvents))
-  }
-
-  private fun startSync(events: Observable<UiEvent>): Observable<UiChange> {
-    val lastSyncedStateStream = lastSyncState
-        .asObservable()
-        .distinctUntilChanged()
-
-    return events
-        .ofType<SyncIndicatorViewClicked>()
-        .withLatestFrom(lastSyncedStateStream)
-        .filter { (_, lastSyncedState) ->
-          lastSyncedState.lastSyncProgress == null || lastSyncedState.lastSyncProgress != SYNCING
-        }
-        .doOnNext { dataSync.fireAndForgetSync(FREQUENT) }
-        .switchMap { showErrorDialogOnSyncError() }
-  }
-
-  private fun showErrorDialogOnSyncError(): Observable<UiChange> {
-    return dataSync
-        .streamSyncErrors()
-        .take(1)
```

Comment:
This `.take(1)` ensures that the error message is only shown once and not multiple times if multiple sync errors occur (it is possible since there are multiple syncs that can fail). Can we investigate and ensure that this behaviour is not lost as part of the migration?

Reply:
This is behaving as expected. The dialog is not shown multiple times here.